### PR TITLE
[jsscripting] Refactor dependency tracking

### PR DESCRIPTION
**Pull request by J-N-K**
This contains the necessary changes related to https://github.com/openhab/openhab-core/pull/3168 and should be merged immediately after to prevent a broken add-ons build.

Signed-off-by: Jan N. Klug <github@klug.nrw>

EDIT by @florian-h05:
Fixes https://github.com/openhab/openhab-addons/issues/12479

### CODE REVIEW

**Comment by florian-h05** [comment]
@J-N-K Can you explain how I can make my local maven build work?

Is is failing with: 

```
[ERROR] Failed to execute goal org.apache.karaf.tooling:karaf-maven-plugin:4.3.7:verify (karaf-feature-verification) on project org.openhab.automation.jsscripting: Feature resolution failed for [openhab-automation-jsscripting/3.4.0.SNAPSHOT]
[ERROR] Message: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=openhab-automation-jsscripting; type=karaf.feature; version=3.4.0.SNAPSHOT; filter:="(&(osgi.identity=openhab-automation-jsscripting)(type=karaf.feature)(version>=3.4.0.SNAPSHOT))" [caused by: Unable to resolve openhab-automation-jsscripting/3.4.0.SNAPSHOT: missing requirement [openhab-automation-jsscripting/3.4.0.SNAPSHOT] osgi.identity; osgi.identity=openhab-runtime-base; type=karaf.feature [caused by: Unable to resolve openhab-runtime-base/3.4.0.SNAPSHOT: missing requirement [openhab-runtime-base/3.4.0.SNAPSHOT] osgi.identity; osgi.identity=openhab-core-model-rule; type=karaf.feature [caused by: Unable to resolve openhab-core-model-rule/3.4.0.SNAPSHOT: missing requirement [openhab-core-model-rule/3.4.0.SNAPSHOT] osgi.identity; osgi.identity=org.openhab.core.model.rule; type=osgi.bundle; version="[3.4.0.202211200312,3.4.0.202211200312]"; resolution:=mandatory [caused by: Unable to resolve org.openhab.core.model.rule/3.4.0.202211200312: missing requirement [org.openhab.core.model.rule/3.4.0.202211200312] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.openhab.core.model.script)(version>=3.4.0)(!(version>=4.0.0)))" [caused by: Unable to resolve org.openhab.core.model.script/3.4.0.202211200311: missing requirement [org.openhab.core.model.script/3.4.0.202211200311] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.openhab.core.automation.module.script.action)(version>=3.4.0)(!(version>=4.0.0)))"]]]]]
[ERROR] Repositories: {
[ERROR]         file:/home/florianh/media/data_m2/git/openhab-addons/bundles/org.openhab.automation.jsscripting/target/feature/feature.xml
[ERROR]         mvn:org.apache.karaf.features/framework/4.3.7/xml/features
[ERROR]         mvn:org.apache.karaf.features/standard/4.3.7/xml/features
[ERROR]         mvn:org.openhab.core.features.karaf/org.openhab.core.features.karaf.openhab-core/3.4.0-SNAPSHOT/xml/features
[ERROR]         mvn:org.openhab.core.features.karaf/org.openhab.core.features.karaf.openhab-tp/3.4.0-SNAPSHOT/xml/features
[ERROR]         mvn:org.ops4j.pax.web/pax-web-features/7.3.25/xml/features
[ERROR] }
[ERROR] Resources: {
[ERROR]         mvn:com.fasterxml.jackson.core/jackson-annotations/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.core/jackson-core/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.core/jackson-databind/2.12.6.1
[ERROR]         mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-cbor/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-xml/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-base/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.12.6
[ERROR]         mvn:com.fasterxml.jackson.module/jackson-module-jaxb-annotations/2.12.6
[ERROR]         mvn:com.fasterxml.woodstox/woodstox-core/6.2.6
[ERROR]         mvn:com.google.code.gson/gson/2.8.9
[ERROR]         mvn:com.google.guava/failureaccess/1.0.1
[ERROR]         mvn:com.google.guava/guava/30.1-jre
[ERROR]         mvn:com.sun.xml.bind/jaxb-osgi/2.3.3
[ERROR]         mvn:com.thoughtworks.xstream/xstream/1.4.19
[ERROR]         mvn:de.jollyday/jollyday/0.5.10
[ERROR]         mvn:io.swagger.core.v3/swagger-annotations/2.1.9
[ERROR]         mvn:io.swagger.core.v3/swagger-core/2.1.9
[ERROR]         mvn:io.swagger.core.v3/swagger-integration/2.1.9
[ERROR]         mvn:io.swagger.core.v3/swagger-jaxrs2/2.1.9
[ERROR]         mvn:io.swagger.core.v3/swagger-models/2.1.9
[ERROR]         mvn:jakarta.annotation/jakarta.annotation-api/2.0.0
[ERROR]         mvn:jakarta.inject/jakarta.inject-api/2.0.0
[ERROR]         mvn:jakarta.jws/jakarta.jws-api/2.1.0
[ERROR]         mvn:jakarta.validation/jakarta.validation-api/2.0.2
[ERROR]         mvn:jakarta.xml.bind/jakarta.xml.bind-api/2.3.3
[ERROR]         mvn:jakarta.xml.ws/jakarta.xml.ws-api/2.3.3
[ERROR]         mvn:javax.annotation/javax.annotation-api/1.2
[ERROR]         mvn:javax.annotation/javax.annotation-api/1.3.2
[ERROR]         mvn:javax.el/javax.el-api/3.0.0
[ERROR]         mvn:javax.mail/mail/1.4.7
[ERROR]         mvn:javax.measure/unit-api/2.1.3
[ERROR]         mvn:javax.servlet/javax.servlet-api/3.1.0
[ERROR]         mvn:net.java.dev.jna/jna-platform/5.11.0
[ERROR]         mvn:net.java.dev.jna/jna/5.11.0
[ERROR]         mvn:org.apache.aries.component-dsl/org.apache.aries.component-dsl.component-dsl/1.2.2
[ERROR]         mvn:org.apache.aries.jax.rs/org.apache.aries.jax.rs.whiteboard/2.0.0
[ERROR]         mvn:org.apache.aries.spec/org.apache.aries.javax.jax.rs-api/1.0.1
[ERROR]         mvn:org.apache.aries.spifly/org.apache.aries.spifly.dynamic.bundle/1.3.2
[ERROR]         mvn:org.apache.aries/org.apache.aries.util/1.1.3
[ERROR]         mvn:org.apache.commons/commons-lang3/3.12.0
[ERROR]         mvn:org.apache.cxf/cxf-core/3.4.5
[ERROR]         mvn:org.apache.cxf/cxf-rt-frontend-jaxrs/3.4.5
[ERROR]         mvn:org.apache.cxf/cxf-rt-rs-client/3.4.5
[ERROR]         mvn:org.apache.cxf/cxf-rt-rs-sse/3.4.5
[ERROR]         mvn:org.apache.cxf/cxf-rt-security/3.4.5
[ERROR]         mvn:org.apache.cxf/cxf-rt-transports-http/3.4.5
[ERROR]         mvn:org.apache.felix/org.apache.felix.cm.json/1.0.6
[ERROR]         mvn:org.apache.felix/org.apache.felix.configadmin.plugin.interpolation/1.2.2
[ERROR]         mvn:org.apache.felix/org.apache.felix.configadmin/1.9.24
[ERROR]         mvn:org.apache.felix/org.apache.felix.configurator/1.0.14
[ERROR]         mvn:org.apache.felix/org.apache.felix.converter/1.0.14
[ERROR]         mvn:org.apache.felix/org.apache.felix.coordinator/1.0.2
[ERROR]         mvn:org.apache.felix/org.apache.felix.fileinstall/3.7.4
[ERROR]         mvn:org.apache.felix/org.apache.felix.inventory/1.0.6
[ERROR]         mvn:org.apache.felix/org.apache.felix.metatype/1.2.4
[ERROR]         mvn:org.apache.felix/org.apache.felix.scr/2.1.30
[ERROR]         mvn:org.apache.felix/org.apache.felix.webconsole.plugins.ds/2.1.0
[ERROR]         mvn:org.apache.felix/org.apache.felix.webconsole.plugins.event/1.1.8
[ERROR]         mvn:org.apache.geronimo.specs/geronimo-jta_1.1_spec/1.1.1
[ERROR]         mvn:org.apache.karaf.features/org.apache.karaf.features.core/4.3.7
[ERROR]         mvn:org.apache.karaf.features/org.apache.karaf.features.extension/4.3.7
[ERROR]         mvn:org.apache.karaf.http/org.apache.karaf.http.core/4.3.7
[ERROR]         mvn:org.apache.karaf.jaas/org.apache.karaf.jaas.boot/4.3.7
[ERROR]         mvn:org.apache.karaf.jaas/org.apache.karaf.jaas.command/4.3.7
[ERROR]         mvn:org.apache.karaf.jaas/org.apache.karaf.jaas.config/4.3.7
[ERROR]         mvn:org.apache.karaf.jaas/org.apache.karaf.jaas.modules/4.3.7
[ERROR]         mvn:org.apache.karaf.kar/org.apache.karaf.kar.core/4.3.7
[ERROR]         mvn:org.apache.karaf.log/org.apache.karaf.log.core/4.3.7
[ERROR]         mvn:org.apache.karaf.scr/org.apache.karaf.scr.management/4.3.7
[ERROR]         mvn:org.apache.karaf.scr/org.apache.karaf.scr.state/4.3.7
[ERROR]         mvn:org.apache.karaf.services/org.apache.karaf.services.eventadmin/4.3.7
[ERROR]         mvn:org.apache.karaf.shell/org.apache.karaf.shell.commands/4.3.7
[ERROR]         mvn:org.apache.karaf.shell/org.apache.karaf.shell.core/4.3.7
[ERROR]         mvn:org.apache.karaf.webconsole/org.apache.karaf.webconsole.http/4.3.7
[ERROR]         mvn:org.apache.karaf.wrapper/org.apache.karaf.wrapper.core/4.3.7
[ERROR]         mvn:org.apache.karaf/org.apache.karaf.event/4.3.7
[ERROR]         mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.javax-inject/1_2
[ERROR]         mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.javax-websocket-api/1.1_1
[ERROR]         mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.saaj-impl/1.4.0_2
[ERROR]         mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.activation-api-1.1/2.9.0
[ERROR]         mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.activation-api-1.2.1/1.2.1_3
[ERROR]         mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.jaxb-api-2.2/2.9.0
[ERROR]         mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.saaj-api-1.4/1.4_2
[ERROR]         mvn:org.apache.sling/org.apache.sling.commons.johnzon/1.2.14
[ERROR]         mvn:org.apache.ws.xmlschema/xmlschema-core/2.2.5
[ERROR]         mvn:org.apache.xbean/xbean-bundleutils/4.21
[ERROR]         mvn:org.apache.xbean/xbean-finder/4.21
[ERROR]         mvn:org.apache.xbean/xbean-reflect/4.21
[ERROR]         mvn:org.bitbucket.b_c/jose4j/0.7.7
[ERROR]         mvn:org.codehaus.woodstox/stax2-api/4.2.1
[ERROR]         mvn:org.eclipse.emf/org.eclipse.emf.common/2.17.0
[ERROR]         mvn:org.eclipse.emf/org.eclipse.emf.ecore.change/2.14.0
[ERROR]         mvn:org.eclipse.emf/org.eclipse.emf.ecore.xmi/2.16.0
[ERROR]         mvn:org.eclipse.emf/org.eclipse.emf.ecore/2.20.0
[ERROR]         mvn:org.eclipse.jdt.core.compiler/ecj/4.6.1
[ERROR]         mvn:org.eclipse.jetty.websocket/javax-websocket-client-impl/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty.websocket/javax-websocket-server-impl/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty.websocket/websocket-api/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty.websocket/websocket-client/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty.websocket/websocket-common/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty.websocket/websocket-server/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty.websocket/websocket-servlet/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-client/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-continuation/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-deploy/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-http/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-io/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-jaas/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-jmx/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-jndi/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-plus/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-proxy/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-rewrite/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-security/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-server/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-servlet/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-servlets/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-util-ajax/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-util/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-webapp/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.jetty/jetty-xml/9.4.46.v20220331
[ERROR]         mvn:org.eclipse.lsp4j/org.eclipse.lsp4j.jsonrpc/0.12.0
[ERROR]         mvn:org.eclipse.lsp4j/org.eclipse.lsp4j/0.12.0
[ERROR]         mvn:org.eclipse.orbit.bundles/com.google.inject/5.0.1.v20210324-2015
[ERROR]         mvn:org.eclipse.orbit.bundles/io.github.classgraph/4.8.138.v20211212-1642
[ERROR]         mvn:org.eclipse.orbit.bundles/org.antlr.runtime/3.2.0.v201101311130
[ERROR]         mvn:org.eclipse.orbit.bundles/org.aopalliance/1.0.0.v201105210816
[ERROR]         mvn:org.eclipse.platform/org.eclipse.equinox.common/3.13.0
[ERROR]         mvn:org.eclipse.platform/org.eclipse.equinox.registry/3.9.0
[ERROR]         mvn:org.eclipse.platform/org.eclipse.equinox.supplement/1.9.0
[ERROR]         mvn:org.eclipse.xtend/org.eclipse.xtend.lib.macro/2.26.0
[ERROR]         mvn:org.eclipse.xtend/org.eclipse.xtend.lib/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.common.types/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.ide/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.smap/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.util/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.xbase.ide/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.xbase.lib/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext.xbase/2.26.0
[ERROR]         mvn:org.eclipse.xtext/org.eclipse.xtext/2.26.0
[ERROR]         mvn:org.fusesource.jansi/jansi/2.4.0
[ERROR]         mvn:org.glassfish.hk2/osgi-resource-locator/1.0.3
[ERROR]         mvn:org.javassist/javassist/3.27.0-GA
[ERROR]         mvn:org.jline/jline/3.21.0
[ERROR]         mvn:org.jmdns/jmdns/3.5.8
[ERROR]         mvn:org.jvnet.staxex/stax-ex/1.7.4
[ERROR]         mvn:org.openhab.addons.bundles/org.openhab.automation.jsscripting/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.addon.marketplace.karaf/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.addon.marketplace/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.audio/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.auth.jaas/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.auth.oauth2client/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.automation.module.media/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.automation.module.script.rulesupport/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.automation.module.script/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.automation.rest/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.automation/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.binding.xml/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.config.core/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.config.discovery.mdns/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.config.discovery/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.config.dispatch/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.config.xml/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.ephemeris/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.id/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.console.karaf/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.console/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.http.auth/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.http/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.jetty.certificate/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.monitor/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.net/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.audio/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.auth/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.core/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.mdns/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.sitemap/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.sse/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.swagger/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.transform/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.ui/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest.voice/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.rest/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.io.transport.mdns/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.karaf/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.core/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.item.ide/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.item.runtime/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.item/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.lsp/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.persistence.ide/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.persistence.runtime/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.persistence/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.rule.ide/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.rule.runtime/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.rule/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.script.ide/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.script.runtime/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.script/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.sitemap.ide/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.sitemap.runtime/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.sitemap/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.thing.ide/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.thing.runtime/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.model.thing/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.persistence/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.semantics/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.storage.json/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.thing.xml/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.thing/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.transform/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.ui.icon/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.ui/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core.voice/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.core.bundles/org.openhab.core/3.4.0-SNAPSHOT
[ERROR]         mvn:org.openhab.osgiify/si.uom.si-units/2.1
[ERROR]         mvn:org.ops4j.pax.logging/pax-logging-api/2.0.16
[ERROR]         mvn:org.ops4j.pax.logging/pax-logging-log4j2/2.0.16
[ERROR]         mvn:org.ops4j.pax.url/pax-url-aether/2.6.10
[ERROR]         mvn:org.ops4j.pax.web/pax-web-api/7.3.25
[ERROR]         mvn:org.ops4j.pax.web/pax-web-extender-whiteboard/7.3.25
[ERROR]         mvn:org.ops4j.pax.web/pax-web-jetty/7.3.25
[ERROR]         mvn:org.ops4j.pax.web/pax-web-jsp/7.3.25
[ERROR]         mvn:org.ops4j.pax.web/pax-web-runtime/7.3.25
[ERROR]         mvn:org.ops4j.pax.web/pax-web-spi/7.3.25
[ERROR]         mvn:org.osgi/org.osgi.service.jaxrs/1.0.0
[ERROR]         mvn:org.osgi/org.osgi.util.function/1.1.0
[ERROR]         mvn:org.osgi/org.osgi.util.function/1.2.0
[ERROR]         mvn:org.osgi/org.osgi.util.promise/1.1.1
[ERROR]         mvn:org.osgi/org.osgi.util.promise/1.2.0
[ERROR]         mvn:org.ow2.asm/asm-analysis/9.2
[ERROR]         mvn:org.ow2.asm/asm-commons/9.2
[ERROR]         mvn:org.ow2.asm/asm-tree/9.2
[ERROR]         mvn:org.ow2.asm/asm-util/9.2
[ERROR]         mvn:org.ow2.asm/asm/9.2
[ERROR]         mvn:org.threeten/threeten-extra/1.5.0
[ERROR]         mvn:org.yaml/snakeyaml/1.27
[ERROR]         mvn:si.uom/si-quantity/2.1
[ERROR]         mvn:tech.units/indriya/2.1.2
[ERROR]         mvn:tech.uom.lib/uom-lib-common/2.1
[ERROR] }
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by J-N-K** [comment]
I guess your local core build does not provide the script actions but the addon-sources require them (recent timer PR). I guess I can rebase my core PR somit included latest core changes, too.

I you don‘t want to wait for that, either checkout both branches from my repo (and first build core, then addons) or fork my repo, rebase the branch on openhab-core/main and build it.

**Review by florian-h05 — COMMENTED** [review]
LGTM, thanks!
I've also tested it, it works.

Please read my comments regarding the `@author` tag.
I know that the number of lines changed is not so high, but there is a bigger change in the functionality/implementation, so I'd add the author tags.

**Comment by florian-h05** [comment]
Fixes https://github.com/openhab/openhab-addons/issues/12479.

**Comment by florian-h05** [comment]
@jpg0 FYI.

**Comment by florian-h05** [comment]
@J-N-K 
Since your core PR has been merged, we should get this merged ASAP.

@jlaur Please see my review above, are the `@author` tags required here? Also see jpg0‘s review above. Are you available for merging this?

File: bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/GraalJSScriptEngineFactory.java [hunk 1]
```diff
@@ -39,13 +41,20 @@
 public final class GraalJSScriptEngineFactory implements ScriptEngineFactory {
```
[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/GraalJSScriptEngineFactory.java:41]: Only required for substantial contributions to the file - see https://www.openhab.org/docs/developer/guidelines.html#c-documentation

[the remark this answers is not in the document]
Comment (florian-h05) [reply bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/GraalJSScriptEngineFactory.java:41]: So not in this case, right?

**Comment by florian-h05** [comment]
Seems like the builds are also failing because we are missing JSScriptServiceUtil from one of my previous PRs, I think rebasing would help.
@J-N-K Do you want to rebase or should I? Seems like I can push to your branch …

**Comment by J-N-K** [comment]
I rebased and fixed the constructor.

**Review by florian-h05 — APPROVED** [review]
Thanks, LGTM!

File: bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/GraalJSScriptEngineFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/GraalJSScriptEngineFactory.java:40]: Correct, not required.

**Review by wborn — APPROVED** [review]
LGTM too! :+1:

**Pull request merged**